MikeSmvl/pawplan
# Firebase Authentication & Features Page

## Overview
This PR introduces Firebase authentication and a new features page to showcase our product capabilities. It includes a complete authentication flow, new UI components, and various infrastructure updates.

## Key Changes
### Authentication
- Implemented Firebase authentication with service worker support
- Added `/auth` page with sign in/sign up functionality
- Created authentication service layer (`/lib/auth.ts`)
- Set up Firebase client and server configurations
- Added auth state management in header component

### Features Page
- Created `/features` page showcasing product capabilities
- Added feature preview images:
  - Behavior Analytics
  - Training Plans
  - Chat Support
  - Training Videos

### UI Components
Added new shadcn/ui components:
- Alert
- Avatar
- Card
- Dropdown Menu
- Label
- Tabs

### Infrastructure
- Updated TSConfig files for Firebase integration
- Removed standalone firebase.json in favor of integrated configuration
- Updated package dependencies
- Configured auth service worker

## Testing Instructions
1. Run `npm install` to get new dependencies
2. Start the development server
3. Test authentication flow at `/auth`
4. Verify protected routes require authentication
5. Check features page at `/features`
6. Verify all UI components render correctly

## Screenshots
![Screenshot 2024-11-17 at 21-06-45 PawPlan](https://github.com/user-attachments/assets/07fce4bd-30af-4dd0-a19d-483efc67fa1d)
![Screenshot 2024-11-17 at 21-06-22 PawPlan](https://github.com/user-attachments/assets/1453328d-0d5f-4771-befb-b7617fc7b0f9)
